### PR TITLE
Save tiles in slide's processed_path

### LINKED ISSUE #114: Tiles generated from a Tiler are not saved inside the "processed_path" of the Slide
By default, tiles generated from the tilers are not saved inside the `processed_path` of the Slide. We are now using the `prefix` attribute of the tiler to point to the same `processed_path`, when it should be used to specify a prefix for the filename or a possible subfolder

### PR DESCRIPTION
**Pull request by alessiamarcolini**
Fix #114

### CODE REVIEW

File: src/histolab/tiler.py [hunk 1]
```diff
@@ -410,7 +413,9 @@ def extract(self, slide: Slide):
         tiles_counter = 0
         for tiles_counter, (tile, tile_wsi_coords) in enumerate(random_tiles):
             tile_filename = self._tile_filename(tile_wsi_coords, tiles_counter)
-            tile.save(tile_filename)
+            root_folder = slide.processed_path
```
Comment (ernestoarbitrio) [inline src/histolab/tiler.py:416]: maybe this assignment is not mandatory
`os.path.join(self.processed_path, tile_filename)` is enough i believe

File: tests/unit/test_tiler.py [hunk 2]
```diff
@@ -297,6 +297,34 @@ def it_knows_its_box_mask_lvl(
...
+
+        random_tiler.extract(slide)
+
+        assert os.path.exists(
+            os.path.join(tmp_path_, "processed", "tile_0_level2_0-10-0-10.png")
+        )
+        assert os.path.exists(
+            os.path.join(tmp_path_, "processed", "tile_1_level2_0-10-0-10.png")
+        )
+
```
Comment (ernestoarbitrio) [inline tests/unit/test_tiler.py:327]: can you add some assertion on the called method? Which args or how many time the method or the property are called? Just to be sure that the function is formally correct. IMHO the only assertions on the path are not completely meaninful.

File: tests/unit/test_tiler.py [hunk 3]
```diff
@@ -541,6 +569,32 @@ def and_doesnt_raise_error_with_wrong_coordinates(self, request, tmpdir):
...
+        grid_tiler.extract(slide)
+
+        assert os.path.exists(
+            os.path.join(tmp_path_, "processed", "tile_0_level2_0-10-0-10.png")
+        )
+        assert os.path.exists(
+            os.path.join(tmp_path_, "processed", "tile_1_level2_0-10-0-10.png")
+        )
+
```
Comment (ernestoarbitrio) [inline tests/unit/test_tiler.py:597]: also here

**Comment by alessiamarcolini** [comment]
The tiles should be saved in a subfolder of slide's processed path called "tiles"

**Pull request merged**